js/apple2io.ts

```diff
@@ -3,17 +3,12 @@ import { Card, Memory, MemoryPages, TapeData, byte, Restorable } from './types';
 import { debug, garbage } from './util';
 import { VideoModes } from './videomodes';
 
-type slot = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7;
-type button = 0 | 1 | 2;
-type paddle = 0 | 1 | 2 | 3;
-type annunciator = 0 | 1 | 2 | 3;
-
-interface Annunciators {
-    0: boolean;
-    1: boolean;
-    2: boolean;
-    3: boolean;
-}
+export type slot = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7;
+export type button = 0 | 1 | 2;
+export type paddle = 0 | 1 | 2 | 3;
+export type annunciator = 0 | 1 | 2 | 3;
+
+type Annunciators = Record<annunciator, boolean>;
```

Comment:
When I saw this, I wondered whether there was a practical difference with the previous and, after [playing with it in the TypeScript playground](https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true&target=7&useUnknownInCatchVariables=true&exactOptionalPropertyTypes=true&noImplicitOverride=true&noPropertyAccessFromIndexSignature=true#code/C4TwDgpgBAhgdnArnAxgSxsA9gJygXigAYoAfKARjKgCZqBmAbgChnRIoBBBZdTXAM5VCAJQgpcAEwA88JKgzYcAGigAjLFgA2EeAD4WzNHGAQcAMxgpo3eXyUC6Ab2ZQ3xAFzrNO+C3eUXhraunD+7jRBPqHhbvRRIX7MAL6s7NAA0pzCUADWECBY5lw8Cvw4QizpUFl0hPmFxba8ioI0htUAogCOiDBaAtIAKhSqQzR6BFCdAB4oWoiSEMOjUOOTEDOmcJICUHAQAG5mUAD8UAAUs-OLy+NjFBtbEDt7B8d458A4iNBe72YAJRQf5HMyGCRwATAKDACheHp9AbSLKrWqTQjfX4sSHQ2GRaa9fqDVGqOQtcoY2E-CAsIA), there is not. The only difference I found is that the displayed type for `keyof Annunciators` as you have it here is `0 | 1 | 2 | 3` whereas for the interface it is `keyof Annunciators`.

That said, I'm fine with this as it captures the intent better.